manual/src/server/sal.md

```diff
@@ -0,0 +1,789 @@
+# SAL
+
+## Tfgrid
+
+### Load
+- loads key
+- chooses network
+- starts session and creates state for client
+
+```
+func (c *Client) Load(args Load) error
+```
+
+### DeployVM
+- Deploy a single VM 
+- Choose node id, farm id, etc 
+
+```
+func (c *Client) DeployVM(args tfgridBase.VMDeployment) (tfgridBase.VMDeployment, error)
+```
+
+### GetVMDeployment
+- Retrieve information about the deployed vm
+
+```
+func (c *Client) GetVMDeployment(name string) (tfgridBase.VMDeployment, error)
+```
+
+### CancelVMDeployment
+- Cancel the vm deployment
+
+```
+func (c *Client) CancelVMDeployment(name string) error
+```
+
+### DeployNetwork
+- Deploy VMs that will be in the same network
+- Provide name and network specs
+- Provide the VM specs (on what node, farmid, flist, etc.)
+
+```
+func (c *Client) DeployNetwork(args tfgridBase.NetworkDeployment) (tfgridBase.NetworkDeployment, error)
+```
+
+### GetNetworkDeployment
+- Retrieve information about the vms you deployed by providing the name when deploying
+```
+func (c *Client) GetNetworkDeployment(name string) (tfgridBase.NetworkDeployment, error)
+```
+
+### CancelNetworkDeployment
+- Delete all deployed VMs that are part of the model that goes by the modelName
+
+```
+func (c *Client) CancelNetworkDeployment(name string) error
+```
+
+### ExtendNetworkDeployment
+- Add a VM to the network
+```
+func (c *Client) ExtendNetworkDeployment(args tfgridBase.ExtendNetworkDeployment) (tfgridBase.NetworkDeployment, error)
+```
+
+### ShrinkNetworkDeployment
+- Remove a VM from the network
+```
+func (c *Client) ShrinkNetworkDeployment(args tfgridBase.ShrinkNetworkDeployment) (tfgridBase.NetworkDeployment, error)
+```
+
+### DeployK8SCluster
+- Deploys K8S cluster
+
+```
+func (c *Client) DeployK8SCluster(args tfgridBase.K8SCluster) (tfgridBase.K8SCluster, error)
+```
+
+### GetK8SCluster
+- Retrieve information about the K8S cluster
+
+```
+func (c *Client) GetK8SCluster(args tfgridBase.GetK8SCluster) (tfgridBase.K8sCluster, error)
+```
+
+### CancelK8S
+- Cancel K8S cluster
+
+```
+func (c *Client) CancelK8SCluster(name string) error 
+```
+
+### ExtendK8S
+- Extend the K8S cluster with an extra worker
+- Provide what you need to add the worker (node id, farm id, etc)
+
+```
+func (c *Client) ExtendK8SCluster(args tfgridBase.ExtendK8S) (tfgridBase.K8sCluster, error) 
+```
+
+### ShrinkK8S
+
+```
+func (c *Client) ShrinkK8SCluster(args tfgridBase.ShrinkK8S) (tfgridBase.K8sCluster, error)
+```
+
+### DeployZDB
+- Deploy a Zero DB
```

Comment:
0-DB is the name we use

Reply:
Yes but Deploy0DB would be strange no?

Reply:
Yes Indeed, I meant in this line `Deploy a Zero DB` to be `Deploy a 0-DB or ZDB`